cartographer/mapping_2d/pose_graph/constraint_builder.cc

```diff
@@ -287,6 +307,29 @@ void ConstraintBuilder::DeleteScanMatcher(const mapping::SubmapId& submap_id) {
   submap_scan_matchers_.erase(submap_id);
 }
 
+void ConstraintBuilder::RegisterMetrics(metrics::FamilyFactory* factory) {
+  auto* counts = factory->NewCounterFamily(
+      "/mapping_2d/pose_graph/constraint_builder/constraints",
+      "Constraints computed");
+  kConstraintsSearchedMetric =
+      counts->Add({{"search_region", "local"}, {"matcher", "searched"}});
+  kConstraintsFoundMetric =
+      counts->Add({{"search_region", "local"}, {"matcher", "found"}});
+  kGlobalConstraintsSearchedMetric =
+      counts->Add({{"search_region", "global"}, {"matcher", "searched"}});
+  kGlobalConstraintsFoundMetric =
+      counts->Add({{"search_region", "global"}, {"matcher", "found"}});
+  auto* queues = factory->NewGaugeFamily(
+      "/mapping_2d/pose_graph/constraint_builder/queues", "Queue lengths");
```

Comment:
.../queue_length would be more idiomatic. "queues" sounds like you're monitoring multiple queues.

Reply:
Done